src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/SwitchStatementHighlighter.cs

```diff
@@ -41,35 +44,39 @@ internal class SwitchStatementHighlighter : AbstractKeywordHighlighter<SwitchSta
         /// Finds all breaks and continues that are a child of this node, and adds the appropriate spans to the spans
         /// list.
         /// </summary>
-        private void HighlightRelatedKeywords(SyntaxNode node, List<TextSpan> spans)
+        private void HighlightRelatedKeywords(SyntaxNode node, List<TextSpan> spans, bool highlightBreaks, bool highlightGotos)
         {
-            switch (node)
-            {
-                case BreakStatementSyntax breakStatement:
-                    spans.Add(breakStatement.BreakKeyword.Span);
-                    spans.Add(EmptySpan(breakStatement.SemicolonToken.Span.End));
-                    break;
+            Debug.Assert(highlightBreaks || highlightGotos);
 
-                case GotoStatementSyntax gotoStatement:
-                    var start = gotoStatement.GotoKeyword.SpanStart;
-                    var end = !gotoStatement.CaseOrDefaultKeyword.IsKind(SyntaxKind.None)
-                        ? gotoStatement.CaseOrDefaultKeyword.Span.End
-                        : gotoStatement.GotoKeyword.Span.End;
+            if (highlightBreaks && node is BreakStatementSyntax breakStatement)
+            {
+                spans.Add(breakStatement.BreakKeyword.Span);
+                spans.Add(EmptySpan(breakStatement.SemicolonToken.Span.End));
+            }
+            else if (highlightGotos && node is GotoStatementSyntax gotoStatement
+                && (!gotoStatement.IsKind(SyntaxKind.GotoStatement) || gotoStatement.Expression.IsMissing))
```

Comment:
this expression is too hard to read.  Please take the ``` && (!gotoStatement.IsKind(SyntaxKind.GotoStatement) || gotoStatement.Expression.IsMissing))``` part out of the else-if clause and just do the test inside the block body.  Also, document what this code is trying to accomplish.

Reply:
fixed